addons/uom/models/uom_uom.py

```diff
@@ -166,24 +55,17 @@ def _onchange_critical_fields(self):
 
     @api.model_create_multi
     def create(self, vals_list):
-        for values in vals_list:
-            if 'factor_inv' in values:
-                factor_inv = values.pop('factor_inv')
-                values['factor'] = factor_inv and (1.0 / factor_inv) or 0.0
-        res = super().create(vals_list)
-        res._check_category_reference_uniqueness()
-        return res
-
-    def write(self, values):
-        if 'factor_inv' in values:
-            factor_inv = values.pop('factor_inv')
-            values['factor'] = factor_inv and (1.0 / factor_inv) or 0.0
-
-        res = super().write(values)
-        if ('uom_type' not in values or values['uom_type'] != 'reference') and\
-                not self.env.context.get('allow_to_change_reference'):
-            self._check_category_reference_uniqueness()
-        return res
+        for vals in vals_list:
+            if not vals.get('relative_uom_id', False) and vals.get('relative_factor', 1.0) != 1.0:
```

Comment:
You should also block factor in this case. Also in the discussion relative_uom was optional and we didn't want to block this case